### PR TITLE
Improve check_icons

### PR DESCRIPTION
**Pull request by fbeutin-ledger**

### CODE REVIEW

**Comment by fbeutin-ledger** [comment]
Passes on boilerplate
https://github.com/LedgerHQ/app-boilerplate/actions/runs/4254703178/jobs/7401323166

**Review by xchapron-ledger — DISMISSED** [review]

File: scripts/check_icons.sh [hunk 1]
```diff
@@ -2,44 +2,83 @@
...
 
-    if ! echo "$content" | grep -q "Colors: 2"; then
-        echo "ERROR => Icon should have only 2 colors"
-        error=1
+    return "$error"
+)
+
+check_glyph() (
+    error=0
+    file="$1"
```
Comment (xchapron-ledger) [inline scripts/check_icons.sh:41]: Can you add a log here with something like:
`"Checking glyph file $file"`

File: scripts/check_icons.sh [hunk 2]
```diff
@@ -2,44 +2,83 @@
...
+        fi
     fi
 
     if [[ error -eq 0 ]]; then
-        echo "SUCCESS => Icon \"$file\" is compliant"
+        echo "SUCCESS => Glyph '$file' is compliant"
     else
         echo "FAILURE =>"
-        echo "identify -verbose \"$file\""
+        echo "identify -verbose '$file'"
```
Comment (xchapron-ledger) [inline scripts/check_icons.sh:81]: I think this is a bit to verbose. Maybe we can replace it with: "check identify -verbose ..."

File: scripts/check_icons.sh [hunk 3]
```diff
@@ -66,32 +105,85 @@ check_is_not_boilerplate_icon() (
...
+    sed -i "/^include/d" "$tmp_makefile"
+    sed -i "/^\$(info/d" "$tmp_makefile"
+    sed -i "/^\$(error/d" "$tmp_makefile"
+
+    echo "include $tmp_makefile" > "$repo/Makefile_dumper.mk"
+    echo "dump_ICONNAME:" >> "$repo/Makefile_dumper.mk"
+    echo -e "\t@echo \$(ICONNAME)" >> "$repo/Makefile_dumper.mk"
+
+    target_name="TARGET_${device_name^^}"
+    icon=$(make BOLOS_SDK="none" TARGET_NAME="$target_name" --no-print-directory -C "$repo" -f "Makefile_dumper.mk" dump_ICONNAME)
```
Comment (xchapron-ledger) [inline scripts/check_icons.sh:126]: This doesn't handle icon for variants if any. But I guess this is a great first step

File: scripts/check_icons.sh (hunk 1 above)
Comment (fbeutin-ledger) [reply scripts/check_icons.sh:41]: Done

File: scripts/check_icons.sh (hunk 2 above)
Comment (fbeutin-ledger) [reply scripts/check_icons.sh:81]: ok

File: scripts/check_icons.sh (hunk 3 above)
Comment (fbeutin-ledger) [reply scripts/check_icons.sh:126]: True. TBH I would like to export the constraint that icons should be placed in `app-xyz/icons` and be done with this mess checking

**Review by xchapron-ledger — APPROVED** [review]

**Pull request merged**